portfolio/src/main/webapp/script.js

```diff
@@ -98,28 +104,53 @@ function showBlogPost(postNum) {
 function hideBlogPost(postNum) {
   // Find the correct triangle to hide and switch its image.
   // Also change the onclick function for this button to show the post.
-  const buttonId = "expandTriangle" + postNum;
-  let button = document.getElementById(buttonId);
-  console.log(button);
+  const buttonId = 'expandTriangle' + postNum;
+  const button = document.getElementById(buttonId);
   if (button !== null) {
-    button.src = "/images/righttriangle.jpeg";
+    button.src = '/images/righttriangle.jpeg';
     button.onclick = function() {
       showBlogPost(postNum);
     };
-  }  
+  }
 
   // Remove blog post content from its area.
-  const blogPostId = "blogPostArea" + postNum;
-  let postArea = document.getElementById(blogPostId);
+  const blogPostId = 'blogPostArea' + postNum;
+  const postArea = document.getElementById(blogPostId);
   if (postArea !== null) {
     if (postNum > 0 && postNum <= posts.length) {
-      const elements = posts[postNum-1]
-      for (i = 0; i < elements.length; i++) {
-        const element = elements[i]
+      const elements = posts[postNum - 1];
+      let i = 0;
+      for (i; i < elements.length; i++) {
+        const element = elements[i];
         if (element !== null) {
           postArea.removeChild(element);
         }
       }
     }
   }
 }
+
+/**
+ * Another way to use fetch is by using the async and await keywords. This
+ * allows you to use the return values directly instead of going through
+ * Promises.
+ */
+async function getCommentsUsingAsyncAwait() {
+  const response = await fetch('/data');
+  const comments = await response.json();
```

Comment:
I imagine you shouldn't need to wait once you have your response. The response should include the json, right?

Reply:
I stand corrected: https://www.reddit.com/r/javascript/comments/5cg27o/why_does_fetch_with_es7_syntax_require_two_await/

So the reason is that this explicitly captures that the headers arrive first. Ignore my suggestion from before.